Read text file

In [1]:
import os
os.chdir('../data')
text = open('nineties.txt').read()

Functions:

In [2]:
from collections import Counter
import os
import string
import json
import matplotlib.pyplot as plt
from IPython.display import Image

In [3]:
import os
os.chdir('../data_analysis')
%run functions.ipynb

### 1990s corpus set up 
* (eliminate toc, create list of docs, tokenize each, flatten)

In [54]:
text.index("200. AM-Atlantic-Today, ; hqq; HALIFAX OUT; UNDATED - 416 . . . Society.; Dartmouth water rates set to rise")

11214

In [55]:
text.index('\nWarner Lambert Recalls Eight Drugs\nT')

11325

In [56]:
text[11214:11325]

'200. AM-Atlantic-Today, ; hqq; HALIFAX OUT; UNDATED - 416 . . . Society.; Dartmouth water rates set to rise\n\n\n\n'

In [57]:
notoc_text=text[11325:]

In [58]:
doc_list=notoc_text.split('End of Document')
len(doc_list)

201

In [59]:
token_docs=[]
for doc in doc_list:
    doc_tokens=tokenize(doc,lowercase=True,strip_chars='!,."')
    token_docs.append(doc_tokens)

In [60]:
flat_list=[]
for sublist in token_docs:
    flat_list.extend(sublist)

len(flat_list)

86407

* pointer for flattened, normalized and tokenized list of total words across documents: "flat_list"

Filter out stop words:

In [45]:
from nltk.corpus import stopwords
stoplist=stopwords.raw('english').split('\n')

In [46]:
filtered_flat_list=[]
for words in flat_list:
    if words not in stoplist:

        filtered_flat_list.append(words)

* pointer: "filtered_flat_list"

## Frequency Lists

Top 100 words

In [28]:
counted_toks=Counter(filtered_flat_list)
print(counted_toks.most_common(100))

[('drug', 1044), ('said', 1042), ('drugs', 570), ('recall', 445), ('company', 416), ('fda', 323), ('patients', 279), ('recalled', 269), ('diet', 234), ('body', 233), ('products', 222), ('press', 204), ('copyright', 204), ('words', 203), ('length:', 200), ('million', 194), ('redux', 194), ('new', 183), ('one', 181), ('used', 178), ('associated', 176), ('section:', 171), ('people', 169), ('load-date:', 167), ('would', 165), ('health', 162), ('1997', 159), ('two', 155), ('market', 155), ('blood', 151), ('news', 149), ('-', 147), ('could', 143), ('heart', 141), ('product', 136), ('food', 135), ('year', 134), ('also', 131), ('doctors', 131), ('$', 131), ('percent', 130), ('dateline:', 126), ('may', 126), ('last', 125), ('fenfluramine', 125), ('sales', 124), ('pharmaceutical', 123), ('american', 118), ('administration', 117), ('home', 110), ('inc', 106), ('medical', 100), ('byline:', 100), ('fen-phen', 100), ('since', 98), ('1998', 98), ('weight', 97), ('rights', 95), ('september', 95), ('oc

Top 50 bigrams

In [29]:
bigrams=get_ngram_tokens(flat_list,n=2)
count_bigrams=Counter(bigrams)
print(count_bigrams.most_common(50))

[('of the', 534), ('in the', 327), ('the drug', 309), ('the company', 263), ('the fda', 223), ('said the', 190), ('the drugs', 182), ('the recall', 180), ('to the', 172), ('associated press', 164), ('on the', 162), ('and drug', 126), ('that the', 124), ('from the', 122), ('for the', 122), ('in a', 121), ('food and', 119), ('by the', 117), ('he said', 114), ('drug administration', 109), ('it was', 108), ('of a', 107), ('the market', 107), ('and the', 106), ('with the', 102), ('to be', 100), ('words byline:', 99), ('redux and', 93), ('the food', 92), ('all rights', 85), ('rights reserved', 85), ('have been', 83), ('diet drugs', 82), ('had been', 81), ('new york', 79), ('american home', 77), ('news length:', 76), ('at the', 76), ('recall of', 75), ('words dateline:', 71), ('of its', 69), ('press all', 65), ('after the', 64), ('it is', 63), ('the product', 62), ('reserved section:', 61), ('said it', 61), ('has been', 59), ('sales of', 57), ('cycle copyright', 56)]


Top 50 trigrams

In [30]:
trigrams=get_ngram_tokens(flat_list,n=3)
count_tri=Counter(trigrams)
print(count_tri.most_common(50))

[('food and drug', 118), ('and drug administration', 106), ('the food and', 91), ('all rights reserved', 85), ('associated press all', 65), ('press all rights', 65), ('rights reserved section:', 61), ('of the drug', 52), ('american home products', 49), ('the associated press', 47), ('the company said', 43), ('words byline: by', 37), ('the fda said', 36), ('section: business news', 35), ('business news length:', 35), ('from the market', 35), ('the united states', 35), ('redux and fenfluramine', 34), ('the diet drugs', 32), ('dateline: washington body', 31), ('the recall of', 31), ('on the market', 31), ('am cycle copyright', 29), ('said the company', 29), ('the new york', 29), ('in the united', 29), ('one of the', 27), ('eastern time copyright', 27), ('the drug was', 26), ('approved by the', 25), ('off the market', 24), ('used to treat', 24), ('heart valve damage', 23), ('by the fda', 23), ('redux and pondimin', 23), ('drug administration said', 22), ('dateline: new york', 22), ('drugs 

### Observations:

Top Words (tokens):
* "Drug" and "said" are at the top, showing a strong focus on drugs and reporting statements
* Lots of mentions of "recall," "recalled," and "company," signaling ongoing product crises and corporate responses
* The appearance of "patients," "doctors," and "medical" shows a shift toward health impacts and clinical concerns
* Words like "fen-phen," "redux," and "fenfluramine" suggest a big focus on weight-loss drugs and controversies related to that
* "FDA" and "administration" are still important but show up less than in the 1980s, maybe indicating a different regulatory narrative
* Financial and numerical terms like "million," "$," and "percent" highlight the economic scale and stakes involved
* "Market," "sales," and "business" suggest attention to the corporate and financial implications of these crises
* Words like "problems," "side," "effects," and "heart" hint at safety concerns and health risks being at the forefront

Top 50 Bigrams:
* "The drug" (309), "the company" (263), and "the FDA" (223) stand out, highlighting the core focus on drugs, corporations, and regulatory bodies
* "The recall" (180) and "recall of" (75) confirm that recalls are a theme in these documents
* "Food and" (119) and "drug administration" (109) reinforce the FDA's critical role in the narratives
* "Redux and" (93) and "diet drugs" (82) link closely to weight-loss drug controversies of the 1990s
* "Associated press" (164) and "words byline:" (99) show the media’s significant role in shaping the stories
* Phrases like "he said" (114) and "said it" (61) suggest these documents heavily rely on quotes and statements
* "The market" (107) and "sales of" (57) emphasize the financial and business impact of the crises
* "American home" (77) is notable and ties back to key players in the pharmaceutical space like Wyeth-Ayerst
* Legal phrases like "all rights" (85) and "rights reserved" (85) point to formal and structured reporting styles


Top 50 Trigrams:
* "Food and drug," "and drug administration," and "the food and" highlight the FDA's central role 
* Legal phrases like "all rights reserved," "associated press all," and "press all rights" reflect formal and structured reporting styles
* "American home products" points to key players in the pharmaceutical industry during this period
* Trigrams like "the company said" and "said the company" show a heavy reliance on corporate statements
* "Redux and fenfluramine" and "redux and pondimin" connect to the weight-loss drug controversies dominating the 1990s
* "The recall of" and "off the market" reinforce the recurring focus on product recalls and market withdrawals
* Health impacts like "heart valve damage" reflect the significant risks associated with certain drugs
* "The fda said" and "by the fda" underscore the critical role of regulatory communication in shaping public perception

J&J word list:

In [31]:
jj_words=['showed','safety','responsible','trust', 'trusted','forgiveness','consumer','example','rectification','sympathy']

In [32]:
words_90s=[]
for word in filtered_flat_list:
    if word in jj_words:
        words_90s.append(word)

len(words_90s)

124

* only 124 instances combined total of these words occur within a corpus of 86,000+ words

In [33]:
count_word=Counter(words_90s)
count_word

Counter({'safety': 59,
         'consumer': 26,
         'showed': 22,
         'example': 5,
         'responsible': 9,
         'trusted': 2,
         'sympathy': 1})

* 7 of the 10 words appear within the entire corpus

## Context Analysis using KWIC concordances:

#### 1. "Safety"

In [21]:
safety=make_kwic('safety',flat_list,win=6)
print_kwic(sort_kwic(safety))

              drug administration the recall is not  safety  related but stems from the validation
                saying it was important for patient  safety  that the letters were acted on
             the necessary steps to protect patient  safety  he said pratt said the ministry
                        march 24 1989 spill and the  safety  of other oil industry operations in
       needed to ensure compliance with operational  safety  emergency response and environmental requirements that's
                            it had no evidence of a  safety  problem with the drug interpharm was
                       are not aware of any serious  safety  problems with the pbi drug and
                      recall was not for health and  safety  reasons he said the drugs in
                      recall was not for health and  safety  reasons he said the drugs in
                     the recalls do not involve any  safety  concerns and do not extend to
               loss discovery of the cen

* "manufacturer abbott inc had upgraded its  safety  methods to guard against future problems"; such instances showing how companies taking more and more initiative/positive measures
* shows more words like "review" (of products, in context)
* a little more ambiguity surrounding direct action by companies (i.e. statements like "need more time to evaluate")
* still loaded with many instances involving "fda criticized" or raised safety concerns
  * there is one statement in which it was claimed the fda "hadn't" criticized the efficacy of the drug in question
* also some statements claiming that recall was "not for health and safety reasons"
* more concerns regarding the FDA approval of the drug, rather than the manufacturer/company cheating safety protocols

#### 2. "Consumer"

In [54]:
consumer=make_kwic('consumer',flat_list,win=6)
print_kwic(sort_kwic(consumer))

          manufactures and markets health care and  consumer  products drug recall usa today june
             be returned load-date: january 6 1994  consumer  alert: drug recall pr newswire november
              initiated a nationwide recall to the  consumer  level retail pharmacists are being requested
                    their pharmacist or may call a  consumer  information line at 800-438-2000 contact: harold
             and marketing quality health care and  consumer  products it employs approximately 34000 contact:
                   october 22 1997 study shows low  consumer  recall of advertised drugs medical marketing
              weight control not advertised to the  consumer  and withdrawn from the market was
             483 words byline: david dishneau body  consumer  advocates urged the food and drug
         said they have already enacted sufficient  consumer  protections including urging doctors to consider
          sales resume jan 25 the washington-based  consumer 

* "enacted sufficient  consumer  protections including urging doctors to consider"
* "guard against future problems for the consumer"
* "weight control not advertised to the  consumer and withdrawn from the market"-here shows some lack of consumer consideration (misinformation on bottle)
* Often mentioned in the context of "consumer safety" or "consumer protection," showing a focus on public health
* Used in phrases like "consumer trust" and "consumer confidence," reflecting the importance of maintaining reputation during crises


#### 3. "Showed"

In [46]:
showed=make_kwic('showed',flat_list,win=6)
print_kwic(sort_kwic(showed))

          its blood products after an autopsy  showed  a blood donor with a rare
               a male donor who died recently  showed  he had creutzfelt-jakob disease a virus
   england journal of medicine later findings  showed  that of 291 patients screened 30
  harvard medical school released in february  showed  that prozac created intense violent suicidal
      medical school and released in february  showed  that prozac crated intense violent suicidal
 patients had abnormal readings although they  showed  no symptoms earlier the fda had
          had a heart murmur subsequent tests  showed  that one or more heart valves
                 said all of the studies that  showed  drugs worked also included diet and
          medicine'' she said one small study  showed  that nearly one-third of people taking
                  go out and play football he  showed  his ability against aston villa and
 sold to unwary pharmacies investigation also  showed  that counterfeit bottles labelled 

* "company tests  showed  they were of a relatively harmless"-ambiguous, potentially negative
* "said all of the studies that  showed  drugs worked"-similarly ambiguous, and possibly negative connotation
* Often used to highlight test results or studies, emphasizing evidence-based claims
* Frequently associated with health risks, like heart valve issues or microbial growth, tying back to consumer safety concerns
* Used ambiguously in phrases like "company tests showed they were of a relatively harmless" and "said all of the studies that showed drugs worked," which could carry a negative connotation depending on the context


#### 4. "Example"

In [59]:
example=make_kwic('example',flat_list,win=6)
print_kwic(sort_kwic(example))

 12 others is only one shocking  example  cchr members handed out the recent
certain foods or bee stings for  example  users inject themselves with epinephrine also
certain foods or bee stings for  example  users inject themselves with epinephrine also
   the fda knew that duract for  example  caused serious liver damage if taken
   makes duract you see here an  example  of the system working documents show


* "FDA knew for example caused serious liver damage"-negative
* the term "example" used here in the context of "as an example," or something along those lines

#### 5. "Responsible"

In [48]:
responsible=make_kwic('responsible',flat_list,win=6)
print_kwic(sort_kwic(responsible))

             the company is found to be  responsible  its penalty is unlikely to go
             the company is found to be  responsible  its penalty is unlikely to go
hassner sharav director of citizens for  responsible  care in psychiatry and research the
         they were ever sold whoever is  responsible  for creating the attitude at the
              friedman said it is not a  responsible  position to suggest the agency would
             the company is found to be  responsible  its penalty is unlikely to go
     up the drug-approval process he is  responsible  for overseeing the process karpinski said
             the company is found to be  responsible  its penalty is unlikely to go
    provided by former workers who were  responsible  for the production process michael riley


* responsible used in context of identifying the company in question and noting how they are either being let go of penalties/consequences
* therefore used with mostly negative connotations

#### 6. "Trusted"

In [49]:
trusted=make_kwic('trusted',flat_list,win=6)
print_kwic(sort_kwic(trusted))

that the drug company cannot be  trusted  with the health of americans the
that the drug company cannot be  trusted  with the health of americans the


* Trusted used in a very negative manner (as in, "cannot be trusted")
* Highlights public skepticism and a lack of confidence in corporate responsibility for consumer health

#### 7. "Sympathy"

In [61]:
sympathy=make_kwic('sympathy',flat_list,win=10)
print_kwic(sort_kwic(sympathy))

more dangerous bacteria were wrong he said we have great  sympathy  for the families involved but the company firmly believes that


* shows how company is not taking full responsibility-does extend sympathy to those affected, but then follows with backtracking

## Sentiment Analysis Score:

In [14]:
### Set up:

import os
import re
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from nltk import tokenize
from nltk.sentiment.vader import SentimentIntensityAnalyzer

In [62]:
### Using VADER
sid = SentimentIntensityAnalyzer()

### text set up
nineties_text=notoc_text.lower().replace('\n','')

In [63]:
sid.polarity_scores(nineties_text)

{'neg': 0.094, 'neu': 0.842, 'pos': 0.064, 'compound': -1.0}

### Observation:
* negative compound
* contains a higher negative than positive score; more negative content